src/util/amount.rs

```diff
@@ -1533,6 +1765,149 @@ mod tests {
+                SignedAmount::from_pico(-4_000_000_000),
+            ], // cannot be empty
+            samt3: &[],
+        };
+
+        let expected_with = r#"{"amt1":[1000000000,2000000000],"amt2":[3000000000,4000000000],"amt3":[5000000000,6000000000],"samt1":[-1000000000,-2000000000],"samt2":[-3000000000,-4000000000],"samt3":[-5000000000,-6000000000]}"#;
+        assert_eq!(serde_json::to_string(&with).unwrap(), expected_with);
+
+        let expected_without = r#"{"amt1":[],"amt2":[3000000000,4000000000],"amt3":[],"samt1":[],"samt2":[-3000000000,-4000000000],"samt3":[]}"#;
+        assert_eq!(serde_json::to_string(&without).unwrap(), expected_without);
+    }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn serde_as_pico_vec_deserialize() {
+        use serde_crate::Deserialize;
+
+        #[derive(Deserialize, PartialEq, Debug, Eq)]
+        #[cfg_attr(feature = "serde", serde(crate = "serde_crate"))]
+        struct T {
+            #[serde(
+                default,
+                deserialize_with = "super::serde::as_pico::vec::deserialize_amount"
+            )]
+            pub amt1: Vec<Amount>,
+            #[serde(
+                default,
+                deserialize_with = "super::serde::as_pico::vec::deserialize_amount"
+            )]
+            pub amt2: Vec<Amount>,
+
+            #[serde(
+                default,
+                deserialize_with = "super::serde::as_pico::vec::deserialize_signed_amount"
+            )]
+            pub samt1: Vec<SignedAmount>,
+            #[serde(
+                default,
+                deserialize_with = "super::serde::as_pico::vec::deserialize_signed_amount"
+            )]
+            pub samt2: Vec<SignedAmount>,
+        }
+
+        let t = T {
+            amt1: vec![Amount(1_000)],
+            amt2: vec![],
+            samt1: vec![SignedAmount(-1_000)],
+            samt2: vec![],
+        };
+
+        let t_str = r#"{"amt1": [1000], "amt2": [], "samt1": [-1000], "samt2": []}"#;
+        let t_from_str: T = serde_json::from_str(t_str).unwrap();
+        assert_eq!(t_from_str, t);
+    }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn serde_as_pico_vec_deserialize_negative_amount_error() {
+        use serde_crate::Deserialize;
+
+        #[derive(Deserialize, PartialEq, Debug, Eq)]
+        #[cfg_attr(feature = "serde", serde(crate = "serde_crate"))]
+        struct T {
+            #[serde(
+                default,
+                deserialize_with = "super::serde::as_pico::vec::deserialize_amount"
+            )]
+            pub amt: Vec<Amount>,
+
+            #[serde(
+                default,
+                deserialize_with = "super::serde::as_pico::vec::deserialize_signed_amount"
+            )]
+            pub samt: Vec<SignedAmount>,
+        }
+
+        let t_str = r#"{"amt": [], "samt": [18446744073709551615]}"#;
+        let t: Result<T, serde_json::Error> = serde_json::from_str(t_str);
+        assert!(t.unwrap_err().is_data());
```

Comment:
Add an `assert_eq!()` for `Display` to check the lines in `expecting()` fn that are not covered yet. There is also lines 1908, 2135, 2139.